openassessment/xblock/static/js/spec/lms/oa_staff_area.js

```diff
@@ -557,18 +588,75 @@ describe('OpenAssessment.StaffAreaView', function() {
 
             expect($('.staff__grade__value').text().trim()).toBe("10 Available and 2 Checked Out");
 
+            // Rendering the staff grading form will cause the counts to re-render as well.
+            // Return a different counts template this time to mimic the counts changing.
+            server.staffGradeCountsTemplate = 'oa_staff_grade_learners_count_1.html';
             showInstructorAssessmentForm(staffArea);
 
-            // Render a different staff area teamplate the next time around so counts can update.
-            server.staffAreaTemplate = 'oa_staff_area_full_grading_2.html';
+            expect($('.staff__grade__value').text().trim()).toBe("9 Available and 3 Checked Out");
 
-            // Fill in assessment and make sure the counts re-render.
-            $assessment = getAssessment(staffArea.element);
+            // Fill in assessment and make sure the code re-renders the count form.
+            $assessment = getAssessment(staffArea.element, staffAreaTab);
             fillAssessment($assessment);
-            server.staffGradeFormTemplate = 'oa_staff_grade_learners_assessment_2.html';
-            submitAssessment(staffArea);
+            // Return yet another counts template to mimic the counts changing.
+            server.staffGradeCountsTemplate = 'oa_staff_grade_learners_count_2.html';
+            submitAssessment(staffArea, staffAreaTab);
+
+            expect($('.staff__grade__value').text().trim()).toBe("9 Available and 2 Checked Out");
+        });
+
+        it('warns of unsubmitted assessments', function() {
+            var staffArea = createStaffArea({}, 'oa_staff_area_full_grading.html'),
+                $assessment;
+
+            showInstructorAssessmentForm(staffArea);
+
+            expect(staffArea.baseView.unsavedWarningEnabled()).toBe(false);
+
+            // Fill in assessment and make sure the code re-renders the count form.
+            $assessment = getAssessment(staffArea.element, staffAreaTab);
+            fillAssessment($assessment);
+            expect(staffArea.baseView.unsavedWarningEnabled()).toBe(true);
+
+            submitAssessment(staffArea, staffAreaTab);
+            expect(staffArea.baseView.unsavedWarningEnabled()).toBe(false);
+        });
+
+        it('tracks unsubmitted assessments in multiple views', function() {
```

Comment:
Nit: I'd move this test out of the 'Grade Available Responses' section since it tests cross-view capabilities.